components/AppLayout/AppHeader.js

```diff
@@ -1,64 +1,281 @@
+
+  const openProfileMenu = useCallback(e => setAnchor(e.currentTarget), [
+    anchor,
+  ]);
+  const closeProfileMenu = useCallback(() => setAnchor(null), [anchor]);
+
+  return (
+    <header className={classes.root}>
+      <div className={classes.top}>
+        <div className={classes.flex}>
+          <a href="/">
+            <img
+              className={classes.logo}
+              src={isDesktop ? '/logo-desktop.png' : '/logo-mobile.png'}
+              alt=""
+            />
+          </a>
+          {isDesktop && <Links classes={classes} />}
+        </div>
+        <GlobalSearch />
+        {isDesktop &&
+          (user?.name ? (
+            <>
+              <img
+                className={classes.avatar}
+                src={getGravatar(user.email)}
+                alt=""
+                onClick={openProfileMenu}
+              />
+              <Menu
+                id="profile-menu"
+                classes={{ paper: classes.profileMenu }}
+                anchorEl={anchor}
+                keepMounted
+                open={!!anchor}
+                onClose={closeProfileMenu}
+              >
+                <span className={classes.level}>LV. {user.level}</span>
+                <MenuItem onClick={closeProfileMenu}>
+                  <ListItemIcon>
+                    <img
+                      className={classes.avatar}
+                      src={getGravatar(user.email)}
+                      alt=""
+                    />
+                  </ListItemIcon>
+                  <Typography variant="inherit">{user.name}</Typography>
+                </MenuItem>
+                <Divider classes={{ root: classes.divider }} />
+                <MenuItem onClick={closeProfileMenu}>
+                  <ListItemIcon className={classes.listIcon}>
+                    <AccountCircleOutlinedIcon />
+                  </ListItemIcon>
+                  <Typography variant="inherit">{t`My Profile`}</Typography>
+                </MenuItem>
+                <Divider classes={{ root: classes.divider }} />
+                <MenuItem onClick={closeProfileMenu}>
+                  <ListItemIcon className={classes.listIcon}>
+                    <InfoIcon />
+                  </ListItemIcon>
+                  <Typography variant="inherit">{t`About`}</Typography>
+                </MenuItem>
+                <Divider classes={{ root: classes.divider }} />
+                <MenuItem onClick={logout}>
+                  <ListItemIcon className={classes.listIcon}>
+                    <ExitToAppRoundedIcon />
+                  </ListItemIcon>
+                  <Typography variant="inherit">{t`Logout`}</Typography>
+                </MenuItem>
+              </Menu>
+            </>
+          ) : (
+            <Button
+              onClick={openLoginModal}
+              size="small"
+              variant="medium"
+            >{t`Login`}</Button>
+          ))}
+      </div>
+      {!isDesktop && (
```

Comment:
Let's use `<Box display={...}>...</Box>` or [`<Hidden implementation="css"></Hidden>`](https://material-ui.com/api/hidden/) to replace `useMediaQuery` here, as suggested by [useMediaQuery doc](https://material-ui.com/components/use-media-query/#server-side-rendering).